nixos/lib/eval-config.nix

```diff
@@ -8,53 +8,46 @@
 # as subcomponents (e.g. the container feature, or nixops if network
 # expressions are ever made modular at the top level) can just use
 # types.submodule instead of using eval-config.nix
-{ # !!! system can be set modularly, would be nice to remove
-  system ? builtins.currentSystem
-, # !!! is this argument needed any more? The pkgs argument can
-  # be set modularly anyway.
-  pkgs ? null
-, # !!! what do we gain by making this configurable?
-  baseModules ? import ../modules/module-list.nix
-, # !!! See comment about args in lib/modules.nix
-  extraArgs ? {}
-, # !!! See comment about args in lib/modules.nix
-  specialArgs ? {}
-, modules
-, # !!! See comment about check in lib/modules.nix
-  check ? true
+# !!! All these arguments can be configured in a modular way
+# via the 'configureModules' attribute.
+{ modules
+, baseModules ? import ../modules/module-list.nix
 , prefix ? []
+, extraArgs ? {}
+, specialArgs ? {}
 , lib ? import ../../lib
+, system ? builtins.currentSystem
+, pkgs ? null
+, check ? true
 }:
 
-let extraArgs_ = extraArgs; pkgs_ = pkgs;
-    extraModules = let e = builtins.getEnv "NIXOS_EXTRA_MODULE_PATH";
```

Comment:
Although there is no need anymore for this environment variable, in theory at least, in practice people might still want to use it, so this should probably be kept somehow for backwards compatibility.